web_widget_x2many_2d_matrix/static/src/components/x2many_2d_matrix_renderer/x2many_2d_matrix_renderer.esm.js

```diff
@@ -162,6 +164,12 @@ export class X2Many2DMatrixRenderer extends Component {
             record: record,
             name: this.matrixFields.value,
         };
+        const domain = record.fields[this.matrixFields.value].domain;
+        if (domain) {
+            result.domain = new Domain(
+                evaluateExpr(domain, record.evalContext)
+            ).toList();
+        }
```

Comment:
When there is no domain specified, the domain is an empty list [].

`Boolean([])` return True, which return error at `evaluateExpr(domain, record.evalContext)`

```
Caused by: EvalError: Can not parse python expression: ()
    Error: Missing token
    EvalError: Can not parse python expression: ()
    Error: Missing token
```

Probably `if (domain.length) {` is the solution.

Reply:
Steps to reproduce:
- Test with the field of `field_value` has no domain specified, or domain = [] (domain = "[]" still works)

Reply:
https://github.com/OCA/web/pull/3116

Reply:
@cuongnmtm I have tested it and its working fine even if there is no domain specified